Squared/Threading/Future.cs

```diff
@@ -75,6 +76,7 @@ internal static class FutureHelpers {
             try {
                 thunk.Invoke();
             } catch (System.Reflection.TargetInvocationException ex) {
+                // REVIEW: Which cases is this catch block for?
```

Comment:
Good question, I don't remember. I know Delegate.Invoke always throws TargetInvocationException so that might be why it's doing this unwrap here. It definitely needed a comment.

Reply:
Yeah I have tried to convince `Delegate.DynamicInvoke` to trigger this exception but I mostly succeeded in it raising "other" exception types... (and that it would not make sense to unwrap to the client)